src/Mollie/WC/Helper/ApplePayDirectHandler.php

```diff
@@ -0,0 +1,1180 @@
+            "amount" => $subtotal,
+            "type" => $type
+        ];
+    }
+
+    /**
+     * Creates a response formatted for ApplePay
+     *
+     * @param array $paymentDetails
+     *
+     * @return array
+     */
+    protected function appleFormattedResponse(array $paymentDetails)
+    {
+        $response = [];
+        if ($paymentDetails['shippingMethods']) {
+            $response['newShippingMethods']
+                = $paymentDetails['shippingMethods'];
+        }
+
+        $response['newLineItems'] = $this->appleNewLineItemsResponse(
+            $paymentDetails
+        );
+
+        $response['newTotal'] = $this->appleNewTotalResponse(
+            $paymentDetails['total']
+        );
+        return $response;
+    }
+
+    /**
+     * Creates NewLineItems line
+     *
+     * @param array $paymentDetails
+     *
+     * @return array[]
+     */
+    protected function appleNewLineItemsResponse(array $paymentDetails)
+    {
+        $type = 'final';
+        $response = [];
+        array_push(
+            $response,
+            $this->appleItemFormat(
+                'Subtotal',
+                $paymentDetails['subtotal'],
+                $type
+            )
+        );
+        if ($paymentDetails['shipping']) {
+            array_push(
+                $response,
+                $this->appleItemFormat(
+                    $paymentDetails['shipping']['label'],
+                    $paymentDetails['shipping']['amount'],
+                    $type
+                )
+            );
+        }
+        array_push(
+            $response,
+            $this->appleItemFormat(
+                'Estimated Tax',
+                $paymentDetails['taxes'],
+                $type
+            )
+        );
+        return $response;
+    }
+
+    /**
+     * Returns a success response to be handled by the script
+     *
+     * @param array $response
+     */
+    protected function responseSuccess(array $response)
+    {
+        wp_send_json_success($response);
+    }
+
+    /**
+     * Returns the authorization response with according success/fail status
+     * Adds the error list if provided to be handled by the script
+     * On success it adds the redirection url
+     *
+     * @param        $status 0 => success, 1 => error
+     * @param string $orderId
+     * @param array  $errorList
+     *
+     * @return array
+     */
+    protected function authorizationResultResponse(
+        $status,
+        $orderId = '',
+        $errorList = []
+    ) {
+        $response = [];
+        if ($status === 'STATUS_SUCCESS') {
+            $response['returnUrl'] = $this->redirectOnApplePaySuccessfulPayment(
+                $orderId
+            );
+            $response['responseToApple'] = ['status' => 0];
+        } else {
+            $response = [
+                'status' => 1,
+                'errors' => $this->applePayError($errorList)
+            ];
+        }
+
+        return $response;
+    }
+
+
+}
```

Comment:
To be honest I found this class extremely big where the concerns and responsibilities are not separated. 

I don't know why putting bootstrap, method which add ajax actions, validations, usage of global variables and long methods with nested if.

I would split this code into multiple classes, some of them maybe reusable such as the nonce validation, do the bootstrap somewhere else and do not use `$_POST` at all but `filter_post` etc...